### PR TITLE
Fix issues found by the dependency analysis

### PR DESCRIPTION
**Pull request by rsynek**
Removes XStream from remaining places with one exception - PlannerTestUtils. That will be handled by a separate PR.

### CODE REVIEW

File: optaplanner-core/pom.xml [hunk 1]
```diff
@@ -80,6 +80,10 @@
       <scope>test</scope>
     </dependency>
     <!-- XML -->
+    <dependency>
+      <groupId>com.thoughtworks.xstream</groupId>
+      <artifactId>xstream</artifactId>
+    </dependency>
```
Comment (yurloc) [inline optaplanner-core/pom.xml:86]: This should be either `<scope>test</scope>` or, even better, removed completely.

File: optaplanner-core/src/main/java/org/optaplanner/core/config/solver/SolverConfig.java [hunk 2]
```diff
@@ -112,14 +110,8 @@ public static SolverConfig createFromXmlResource(String solverConfigResource, Cl
...
-        } catch (ConversionException e) {
-            String lineNumber = e.get("line number");
-            throw new IllegalArgumentException("Unmarshalling of solverConfigResource (" + solverConfigResource
-                    + ") fails on line number (" + lineNumber + ")."
-                    + (Objects.equals(e.get("required-type"), "java.lang.Class")
-                            ? "\n  Maybe the classname on line number (" + lineNumber
-                                    + ") is surrounded by whitespace, which is invalid."
-                            : ""),
+        } catch (XmlUnmarshallingException e) {
+            throw new IllegalArgumentException("Unmarshalling of solverConfigResource (" + solverConfigResource + ") failed.",
```
Comment (ge0ffrey) [inline optaplanner-core/src/main/java/org/optaplanner/core/config/solver/SolverConfig.java:114]: I would replace paste tense with present tense: "failed" -> "fails", for consistency.

File: optaplanner-core/pom.xml (hunk 1 above)
Comment (ge0ffrey) [inline optaplanner-core/pom.xml:86]: We still need it, despite the `ConversionException` import removal?

Comment (yurloc) [reply optaplanner-core/pom.xml:86]: Only for PlanningTestUtils.

**Comment by ge0ffrey** [comment]
@rsynek Merged, regardless of comments because it fixes the build.

**Comment by rsynek** [comment]
The remaining concerns are addressed by #846.

**Pull request merged**